### ISSUE TITLE
get_output_probabilites is incorrect

### ISSUE THREAD

**Issue by sohaibimran7** [AISC10-team09/bayesian_reasoning#10]
# Compute log probabilities using softmax over logits.
  # Shift logits and labels to compute the probability of each token given its context.
  shift_logits = logits[:, :-1, :]  # shape: [1, seq_len-1, vocab_size]
  shift_labels = input_ids[:, 1:]     # shape: [1, seq_len-1]

I don't know why we are getting rid of the final logits. The get_output_probabilities function in Aditya's latest PR fixes this and can replace the function here.
